datahub-actions/src/datahub_actions/utils/social_util.py

```diff
@@ -0,0 +1,138 @@
+import datetime
+import logging
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from datahub.ingestion.graph.client import DataHubGraph
+from datahub.metadata.schema_classes import EntityChangeEventClass as EntityChangeEvent
+from datahub.utilities.urns.urn import Urn
+
+from datahub_actions.utils.name_resolver import (
+    get_entity_name_from_urn,
+    get_entity_qualifier_from_urn,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def pretty_any_text(text: str, channel: str) -> str:
+    if text.startswith("http:"):
+        return make_url_with_title(text, text, channel)
+    else:
+        return text
+
+
+def make_url_with_title(title: str, url: str, channel: str) -> str:
+    if channel == "slack":
+        # slack uses mrkdwn format
+        return f"<{url}|{title}>"
+    else:
+        return f"[{title}]({url})"
+
+
+def make_bold(text: str, channel: str) -> str:
+    if not text:
+        return text
+    if channel == "slack":
+        return f"*{text}*"
+    else:
+        return f"**{text}**"
+
+
+@dataclass
+class StructuredMessage:
+    main_message: str
+    properties: Dict[str, str]
+    text: Optional[str]
```

Comment:
the field names here are pretty confusing - main_message is the "sender" and text is the actual message?